packages/jaeger-ui/src/types/config.tsx

```diff
@@ -84,15 +84,15 @@ export type Config = {
     // menuEnabled enables or disables the System Architecture tab.
     menuEnabled?: boolean;
 
-    // dagMaxServicesLen defines the maximum number of services allowed
+    // dagMaxNumServices defines the maximum number of services allowed
     // before the DAG dependency view is disabled. Too many services
     // cause the DAG view to be non-responsive.
-    dagMaxServicesLen?: number;
+    dagMaxNumServices?: number;
   };
 
   // menu controls the dropdown menu in the top-right corner of the UI.
   // When populated, this element completely overrides the default menu.
-  menu: (ConfigMenuGroup | ConfigMenuItem)[];
+  menu: readonly (ConfigMenuGroup | ConfigMenuItem)[];
```

Comment:
curious why these specific entries need `readonly` and not all others.

Reply:
It seems it's specifically for `array` type entries. TS is apparently now better at detecting that the object passed to `deepFreeze` is recursively `readonly`.